packages/zwave-js/src/lib/commandclass/DoorLockLoggingCC.ts

```diff
@@ -0,0 +1,662 @@
+	endpoint: number,
+): ValueID {
+	return {
+		commandClass: CommandClasses["Door Lock Logging"],
+		endpoint,
+		property: "recordsSupported",
+	};
+}
+
+export function getDoorLockLoggingRecordEventTypeID(endpoint: number): ValueID {
+	return {
+		commandClass: CommandClasses["Door Lock Logging"],
+		endpoint,
+		property: "recordEventType",
+	};
+}
+
+export function getDoorLockLoggingRecordNumberID(endpoint: number): ValueID {
+	return {
+		commandClass: CommandClasses["Door Lock Logging"],
+		endpoint,
+		property: "recordNumber",
+	};
+}
+
+export function getDoorLockLoggingRecordTimestampID(endpoint: number): ValueID {
+	return {
+		commandClass: CommandClasses["Door Lock Logging"],
+		endpoint,
+		property: "recordTimestamp",
+	};
+}
+
+export function getDoorLockLoggingRecordUserIDID(endpoint: number): ValueID {
+	return {
+		commandClass: CommandClasses["Door Lock Logging"],
+		endpoint,
+		property: "recordUserID",
+	};
+}
+
+export function getDoorLockLoggingRecordUserCodeID(endpoint: number): ValueID {
+	return {
+		commandClass: CommandClasses["Door Lock Logging"],
+		endpoint,
+		property: "recordUserCode",
+	};
+}
+
+function setRecordMetadata(
+	this: DoorLockLoggingCC,
+	supportedRecordsNumber: number | undefined,
+	userCode?: string | Buffer,
+) {
+	const valueDB = this.getValueDB();
+	const recordEventTypeID = getDoorLockLoggingRecordEventTypeID(
+		this.endpointIndex,
+	);
+	const recordNumberID = getDoorLockLoggingRecordNumberID(this.endpointIndex);
+	const recordTimestampID = getDoorLockLoggingRecordTimestampID(
+		this.endpointIndex,
+	);
+	const recordUserIDID = getDoorLockLoggingRecordUserIDID(this.endpointIndex);
+	const recordUserCodeID = getDoorLockLoggingRecordUserCodeID(
+		this.endpointIndex,
+	);
+
+	// Always create metadata if it does not exist
+	if (!valueDB.hasMetadata(recordTimestampID)) {
+		valueDB.setMetadata(recordTimestampID, {
+			...ValueMetadata.ReadOnlyString,
+			label: `record timestamp`,
+			description: "Timestamp for latest logging record",
+		});
+	}
+
+	const supportedEventTypes = [
+		EventType.LockCode,
+		EventType.UnlockCode,
+		EventType.LockButton,
+		EventType.UnlockButton,
+		EventType.LockCodeOOSchedule,
+		EventType.UnlockCodeOOSchedule,
+		EventType.IllegalCode,
+		EventType.LockManual,
+		EventType.UnlockManual,
+		EventType.LockAuto,
+		EventType.UnlockAuto,
+		EventType.LockRemoteCode,
+		EventType.UnlockRemoteCode,
+		EventType.LockRemote,
+		EventType.UnlockRemote,
+		EventType.LockRemoteCodeOOSchedule,
+		EventType.UnlockRemoteCodeOOSchedule,
+		EventType.RemoteIllegalCode,
+		EventType.LockManual2,
+		EventType.UnlockManual2,
+		EventType.LockSecured,
+		EventType.LockUnsecured,
+		EventType.UserCodeAdded,
+		EventType.UserCodeDeleted,
+		EventType.AllUserCodesDeleted,
+		EventType.MasterCodeChanged,
+		EventType.UserCodeChanged,
+		EventType.LockReset,
+		EventType.ConfigurationChanged,
+		EventType.LowBattery,
+		EventType.NewBattery,
+		EventType.Unknown,
+	];
```

Comment:
You don't need to specify this. Leave out the 2nd argument to `enumValuesToMetadataStates` and `states` will contain all possible values.